frontend/App.js

```diff
@@ -1,28 +1,70 @@
-import { StatusBar } from "expo-status-bar";
-import React from "react";
-import { StyleSheet, Text, View } from "react-native";
+import React, { useState, useRef } from "react";
+import { StyleSheet, View, Text, Animated } from "react-native";
+
 import { registerRootComponent } from "expo";
-import { Provider } from "react-redux";
-import store from "./store/store";
+
+import ActionButton from "./components/MockComponent/ActionButton";
+import ActionDialog from "./components/MockComponent/ActionDialog";
 
 function App() {
+  const [visible, setVisible] = useState(false);
+
+  function cancelActionDialog() {
+    setVisible(false);
+  }
+
+  const dimEffect = useRef(new Animated.Value(0)).current;
+
+  function dimIn() {
+    Animated.timing(dimEffect, {
+      toValue: 1,
+      duration: 1000
+    }).start();
+  }
+
+  function dimOut() {
+    Animated.timing(dimEffect, {
+      toValue: 0,
+      duration: 1000
+    }).start();
+  }
+
+  let dimBackground;
+
+  if (visible) {
+    dimBackground = (
+      <Animated.View style = {[
+        styles.dimContainer, {
+          opacity: dimEffect
+          }
+        ]} 
+      >
+      </Animated.View>
+    );
+  }
+
   return (
-    <Provider store={store}>
```

Comment:
We'll still want this `Provider` component wrapping the return of `App`. If it's breaking the code, ask Gavin about why and he can help you.